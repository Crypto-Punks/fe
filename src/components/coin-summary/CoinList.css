.CoinList {
  position: relative;
  top: 40%;
  left: 23%;
  width: 50%;
  display: flex;
  flex-direction: column;

  a {
    text-decoration: none;
    font-family: abolition, sans-serif;
    font-weight: 400;
    font-style: normal;
  }

  a:visited {
    color: none;
  }

  li {
    position: relative;
    background-color: #ACDEE3;
    margin: 20px;
    padding: 10px;
    height: 16vh;
    border-radius: .8em;
    list-style: none;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    transition: .2s ease-in-out;
    border: solid black;

    img {
      width: 20%;
      height: 100%;
      border-radius: .5em;
    }

    p {
      color: black;
      font-family: abolition, sans-serif;
      font-weight: 400;
      font-style: normal;
    }
    
    .Name {
      left: 0;
      font-size: 20pt;
      letter-spacing: .5em;
      font-weight: bold;
    }

    .Change {
      font-size: 10pt;
    }

    .negative {
      color: rgb(214, 64, 64);
    }

    .positive {
      color: rgb(17, 212, 17);
    }

  }

  li:hover {
    transform: translateY(-0.2em);
    cursor: pointer;
    box-shadow: .5px .5px 35px 1px white;
  }
}

@media (max-width: 800px) {

  .CoinList {
    top: 40%;
    left: 0%;
    width: 80%;

    li {
      height: 10vh;
    }
  }
}

@media (max-width: 500px) {

  .CoinList {
    top: 40%;
    left: 0%;
    width: 80%;

    li {
      height: 8vh;
    }

  }
}