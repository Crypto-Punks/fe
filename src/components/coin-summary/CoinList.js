import React from 'react';
import PropTypes from 'prop-types';
import Coin from './Coin';
import styles from './CoinList.css';

const CoinList = ({ items }) => {
  const elements = items.map(item => {
    return <Coin key={item.name} item={item} />;
  });
  return (
    <ul className={styles.CoinList}>
      {elements}
    </ul>
  );
};

CoinList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    logo: PropTypes.string,
    name: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    changePercent24Hr: PropTypes.string.isRequired
  })).isRequired
};

export default CoinList;
