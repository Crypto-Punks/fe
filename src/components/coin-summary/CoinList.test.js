import React from 'react';
import { shallow } from 'enzyme';
import CoinList from './CoinList';

const data = {
  items: []
};

describe('CoinList component', () => {
  it('renders CoinList', () => {
    const wrapper = shallow(<CoinList { ...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
