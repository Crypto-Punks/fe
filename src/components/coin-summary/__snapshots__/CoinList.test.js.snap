// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinList component renders CoinList 1`] = `
<ul
  className="CoinList"
/>
`;
